gateway/shard/connect.ts

```diff
@@ -8,8 +8,16 @@ export async function connect(shard: Shard): Promise<void> {
   }
   shard.events.connecting?.(shard);
 
-  // Explicitly setting the encoding to json, since we do not support ETF.
-  const socket = new WebSocket(`${shard.gatewayConfig.url}/?v=${shard.gatewayConfig.version}&encoding=json`);
+  const socket = new WebSocket(
+    `${
+      shard.gatewayConfig.url === "wss://gateway.discord.gg"
```

Comment:
@itohatweb do u mind double checking this real quick, everything else in this pr is good just double check this line imo

Reply:
That one looks good.